kivy/uix/recycleview/views.py

```diff
@@ -108,6 +111,34 @@ def apply_selection(self, rv, index, is_selected):
         pass
 
 
+class RecycleKVIDsDataViewBehavior(RecycleDataViewBehavior):
+    """Similar to :class:`RecycleDataViewBehavior`, except that the data keys
+    can signify properties of objects named with an id in KV.
+
+    E.g. given a KV rule::
+
+        <MyRule@RecycleKVIDsDataViewBehavior+BoxLayout>:
+            Label:
+                id: name
+            Label:
+                id: value
+
+    Then setting the data list with
+    ``rv.data = [{'name.text': 'Kivy user', 'value.text': '12'}]`` would
+    automatically set the corresponding labels.
+    """
+
+    def refresh_view_attrs(self, rv, index, data):
+        sizing_attrs = RecycleDataAdapter._sizing_attrs
+        for key, value in data.items():
+            if key not in sizing_attrs:
+                name, *ids = key.split('.', maxsplit=1)
```

Comment:
why the star? the maxsplit makes it a 2-tuple, so we should be able to directly use the value without indexing on the next line.

Also, i think like `ids` is not a great name here, as `name` actually refer to a kv id, and `ids` here would be a chain of one or more attributes.
And why the maxsplit actually? Do we really want to set an attribute with dots in it? wouldn't it be better to iterate on the names, getting the next value until the penultimate one and use setattr on this one?

```python
wid, *attrs = key.split('.')
obj = self
for attr in attrs[:-1]:
    obj = getattr(obj, attr)
setattr(obj, attrs[-1], value)
```
or would that cause issues i can't think of right now?

Reply:
>why the star? the maxsplit makes it a 2-tuple, so we should be able to directly use the value without indexing on the next line.

Because the `maxsplit` is 1 we can have **at most** 2 items. But it could also be only 1. In the latter case, `ids` will be empty.

Reply:
>Also, i think like ids is not a great name here, as name actually refer to a kv id, and ids here would be a chain of one or more attributes.

I feel like that's nitpicking because in the normal case it is the property name, except when it's interpreted as an id :) But also, I think you may have misunderstood what this is doing. Specifically, with the following KV examples:

```
MyRule:
    street: ''
    Label:
        id: address

<RootRule@RecycleKVIDsDataViewBehavior>:
    country: ''
    Label:
        id: phone
    MyRule:
        id: my_rule
```

Setting `data` to `[{'country': 'US', 'phone.text': '123456', 'my_rule.street': 'Home ave', 'my_rule.address.text': '555'}]`. Under the current code, `my_rule` will have a attribute named `address.text` created and set to `555`. That's because we don't recursively go through rules like your example code. We only allow accessing the `ids` at the first level. Hence the only 1 split by period. Under your code, it would set the `text` property of the `address` widget to `555`.

The reason is that I don't think it would be common in kivy to set properties this way for sub rules and sub-sub rules etc. So I don't think we should complicate the API by adding this. Because once we add that it's hard to remove such functionality and I'd prefer to wait until this specifically is requested. And even then I think we can just add such recursive `ids` access a doc recipe. For now, this API only allows setting the property of a widget named in the root rule of the class.

Reply:
ok, i did think a recursive definition could be useful, but i agree it might be a strong commitment that we might not need to do.

Reply:
I suppose we can raise an exception if there is more than one dot, in case users think that it does work recursively.

Reply:
Ok, I added that.